### PR TITLE
Enable multi-api support from lifecycle

### LINKED ISSUE #780: Support for new lifecycle descriptor
### Description

Implement support for the new lifecycle.toml descriptor:

```toml
[apis]
[apis.buildpack]
  deprecated = []
  supported = ["0.2", "0.3", "0.4"]
[apis.platform]
  deprecated = []
  supported = ["0.3", "0.4"]

# For backwards compatibility the minimum supported APIs are provided in RFC-0011 format
# https://github.com/buildpacks/rfcs/blob/main/text/0011-lifecycle-descriptor.md
[api]
  platform = "0.3"
  buildpack = "0.2"

[lifecycle]
  version = "0.9.0"
```

### Proposed solution

If `[apis]` key is present, use solely the information in that table and ignore `[api]`, otherwise expect and use `[api]`.

### Additional context
- [ ] This feature should be documented somewhere

### Tasks

- [x] Read new format
- [x] Set new metadata to builder (incl old fields - [backwards compatibility][backward-compat])
- [x] Read new metadata from builder (falling back to old fields - [backwards compatibility][backward-compat])
- [x] Remove support for platform 0.2 (**breaking-change**)
- [x] Update inspect builder output to include `deprecated` and `supported`
- [x] Build using the latest supported version of the platform API
- [x] ~~Platform API 0.4: [api version in group.toml](https://github.com/buildpacks/spec/pull/120/files)~~
- [ ] Platform API 0.4: [remove version from BOM](https://github.com/buildpacks/spec/issues/107)
- [ ] Platform API 0.4: [removal of CNB_PROCESS_TYPE](https://github.com/buildpacks/rfcs/blob/main/text/0045-launcher-arguments.md#selecting-a-process-type-at-runtime)

Relates to https://github.com/buildpacks/lifecycle/pull/347

[backward-compat]: https://github.com/buildpacks/rfcs/blob/main/text/0049-multi-api-lifecycle-descriptor.md#backwards-compatibility

### PR DESCRIPTION
**Pull request by jromero**
## Summary

This is the initial PR to enable multi-api support. Full support of Platform API 0.4 to follow.

This PR encompasses the following:

- Read new format
- Set new metadata to builder (incl old fields - [backwards compatibility][backward-compat])
- Read new metadata from builder (falling back to old fields - [backwards compatibility][backward-compat])
- Remove support for platform 0.2 (**breaking-change**)
- Update inspect builder output to include `deprecated` and `supported`

## Documentation

- Should this change be documented?
    - [x] Yes, see https://github.com/buildpacks/docs/issues/183
    - [ ] No

## Related

Resolves #780 

[backward-compat]: https://github.com/buildpacks/rfcs/blob/main/text/0049-multi-api-lifecycle-descriptor.md#backwards-compatibility

### CODE REVIEW

File: internal/build/build.go [hunk 1]
```diff
@@ -187,3 +188,19 @@ func randString(n int) string {
 	}
 	return string(b)
 }
+
+func latestVersion(versions []*api.Version) *api.Version {
```
Comment (dfreilich) [inline internal/build/build.go:192]: Considering that this is used twice (for latest version, and the reverse for earliestVersion), is there any utility in changing it slightly to accept a comparison func, so you could call it with something like
```
latestFunc := func(version, latest *api.Version) bool { return version.Compare(latest) > 0 }
appropriateVersion(versions []*api.Version, latestFunc)
```
?

File: internal/builder/builder.go [hunk 2]
```diff
@@ -439,13 +440,22 @@ func validateBuildpacks(stackID string, mixins []string, lifecycleDescriptor Lif
 	for _, bp := range bpsToValidate {
 		bpd := bp.Descriptor()
 
-		if !bpd.API.SupportsVersion(lifecycleDescriptor.API.BuildpackVersion) {
+		// TODO: Should we warn when a buildpack API version is deprecated?
```
Comment (dfreilich) [inline internal/builder/builder.go:443]: 👍

File: testdata/lifecycle/platform-0.4/lifecycle-v0.0.0-arch/creator [hunk 3]
```diff
@@ -0,0 +1 @@
+restorer
```
Comment (dfreilich) [inline testdata/lifecycle/platform-0.4/lifecycle-v0.0.0-arch/creator:1]: nit: 
```suggestion
creator
```

File: internal/build/build.go (hunk 1 above)
Comment (jromero) [reply internal/build/build.go:192]: Great idea! Applied suggestion.

File: acceptance/testdata/pack_fixtures/inspect_image_local_output.txt [hunk 4]
```diff
@@ -8,9 +8,7 @@ LOCAL:
 Stack: pack.test.stack
 
 Base Image:
-{{- if .show_reference}}
```
[the remark this answers is not in the document]
Comment (jromero) [reply acceptance/testdata/pack_fixtures/inspect_image_local_output.txt:11]: Correct. This has been a long time coming: `!l.SemVer().LessThan(semver.MustParse("0.5.0"))`

**Comment by dfreilich** [comment]
woohoo

**Pull request merged**